File: cxo_council/llm_client.py
"""Simple LLM client - uses OpenRouter for all models."""

import os
import httpx


class LLMClient:
    """Minimal LLM client using OpenRouter for CxO Council."""

    def __init__(self):
        self.openrouter_key = os.getenv("OPENROUTER_API_KEY") or os.getenv("COUNCIL_OPENROUTER_API_KEY")

        if not self.openrouter_key:
            raise ValueError("OPENROUTER_API_KEY or COUNCIL_OPENROUTER_API_KEY required")

    def query(self, prompt: str, model: str, temperature: float = 0.7) -> str:
        """Send a query via OpenRouter and get a response."""
        try:
            # Strip openrouter: prefix if present
            model_id = model.replace("openrouter:", "")

            # If it's an anthropic: prefix, convert to OpenRouter format
            if model_id.startswith("anthropic:"):
                model_id = model_id.replace("anthropic:", "anthropic/")

            headers = {
                "Authorization": f"Bearer {self.openrouter_key}",
                "Content-Type": "application/json",
            }

            data = {
                "model": model_id,
                "messages": [{"role": "user", "content": prompt}],
                "temperature": temperature,
            }

            with httpx.Client(timeout=120.0) as client:
                response = client.post(
                    "https://openrouter.ai/api/v1/chat/completions",
                    headers=headers,
                    json=data
                )
                response.raise_for_status()
                result = response.json()
                return result["choices"][0]["message"]["content"]
        except Exception as e:
            return f"ERROR: {str(e)}"

File: cloudflare/shared/llm-client.ts
/**
 * LLM API Client for CxO Council
 * Supports OpenRouter and Anthropic
 */

export class LLMClient {
  constructor(private apiKey: string) {}

  async query(
    prompt: string,
    model: string,
    temperature: number = 0.7
  ): Promise<string> {
    if (model.startsWith('openrouter:')) {
      return this.queryOpenRouter(prompt, model, temperature)
    } else if (model.startsWith('anthropic:')) {
      return this.queryAnthropic(prompt, model, temperature)
    } else if (model.startsWith('openai:')) {
      return this.queryOpenAI(prompt, model, temperature)
    } else {
      throw new Error(`Unknown model provider: ${model}`)
    }
  }

  private async queryOpenRouter(
    prompt: string,
    model: string,
    temperature: number
  ): Promise<string> {
    const modelName = model.replace('openrouter:', '')

    const response = await fetch('https://openrouter.ai/api/v1/chat/completions', {
      method: 'POST',
      headers: {
        'Authorization': `Bearer ${this.apiKey}`,
        'Content-Type': 'application/json',
        'HTTP-Referer': 'https://github.com/yourusername/cxo-council',
      },
      body: JSON.stringify({
        model: modelName,
        messages: [{ role: 'user', content: prompt }],
        temperature
      })
    })

    if (!response.ok) {
      const errorText = await response.text()
      throw new Error(`OpenRouter API error (${response.status}): ${errorText}`)
    }

    const data: any = await response.json()
    return data.choices[0].message.content
  }

  private async queryAnthropic(
    prompt: string,
    model: string,
    temperature: number
  ): Promise<string> {
    const modelName = model.replace('anthropic:', '')

    const response = await fetch('https://api.anthropic.com/v1/messages', {
      method: 'POST',
      headers: {
        'x-api-key': this.apiKey,
        'anthropic-version': '2023-06-01',
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({
        model: modelName,
        max_tokens: 4096,
        messages: [{ role: 'user', content: prompt }],
        temperature
      })
    })

    if (!response.ok) {
      const errorText = await response.text()
      throw new Error(`Anthropic API error (${response.status}): ${errorText}`)
    }

    const data: any = await response.json()
    return data.content[0].text
  }

  private async queryOpenAI(
    prompt: string,
    model: string,
    temperature: number
  ): Promise<string> {
    const modelName = model.replace('openai:', '')

    const response = await fetch('https://api.openai.com/v1/chat/completions', {
      method: 'POST',
      headers: {
        'Authorization': `Bearer ${this.apiKey}`,
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        model: modelName,
        messages: [{ role: 'user', content: prompt }],
        temperature
      })
    })

    if (!response.ok) {
      const errorText = await response.text()
      throw new Error(`OpenAI API error (${response.status}): ${errorText}`)
    }

    const data: any = await response.json()
    return data.choices[0].message.content
  }
}
